Source/diablo.cpp

```diff
@@ -654,6 +654,8 @@ LRESULT GM_Game(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
 
 BOOL LeftMouseDown(int wParam)
 {
+	BOOL shiftIsHeld = wParam == DVL_MK_SHIFT + DVL_MK_LBUTTON;

```

Comment:
You probably want to treat this as a bitflag. Also `BOOL` is just an int, might want to use bool for any new code.
```suggestion
	bool shiftIsHeld = wParam & (DVL_MK_SHIFT | DVL_MK_LBUTTON);
```

Reply:
> You probably want to treat this as a bitflag.

Oh, that looks better indeed. I didn't really pay a lot of attention to this line since I just extracted the existing logic into a variable for reuse.

> Also BOOL is just an int, might want to use bool for any new code.

`bool` for new code it is. Was using `BOOL` because I thought it was some sort of project-wide standard.

Reply:
It was for the original Diablo, but it's actually a WinAPI type that isn't understood by compilers and so gets less error checking and optimizations. I hope to one day replace all cases of `BOOL` (`int`) and `BOOLEAN` (`char`) with `bool` :)